### PR TITLE
Fix to PR #501 - WINDUP-460 RuleMetadata.TREAT_EXCEPTIONS_AS_FATAL

### PR DESCRIPTION
**Pull request by OndraZizka**

### CODE REVIEW

File: config/api/src/main/java/org/jboss/windup/config/metadata/MetadataBuilder.java [hunk 1]
```diff
@@ -444,6 +448,27 @@ public MetadataBuilder addRequiredAddon(AddonId reference)
...
     }
 
+
+    /**
+     * Whether Windup should stop execution if this provider's rule execution ends with an exception.
+     *
+     * By default, the exceptions are only logged and the failing rule appears in report.
+     * The rule itself is responsible for handling exceptions and storing them into the graph.
+     */
+    public void setHaltOnException(boolean haltOnException)
```
Comment (jsight) [inline config/api/src/main/java/org/jboss/windup/config/metadata/MetadataBuilder.java]: Shouldn't this be using the builder pattern?

File: config/api/src/main/java/org/jboss/windup/config/metadata/RuleMetadataType.java [hunk 2]
```diff
@@ -25,13 +27,16 @@
...
 
     /**
-     * Whether or not all Exceptions from this Rule are to be treated as fatal. The default is non-fatal.
+     * Whether Windup should stop execution if this provider's rule execution ends with an exception.
+     *
+     * By default, the exceptions are only logged and the failing rule appears in report.
+     * The rule itself is responsible for handling exceptions and storing them into the graph.
      */
-    TREAT_EXCEPTIONS_AS_FATAL
+    HALT_ON_EXCEPTION
```
Comment (jsight) [inline config/api/src/main/java/org/jboss/windup/config/metadata/RuleMetadataType.java]: I do like this name better! :)

File: config/api/src/main/java/org/jboss/windup/config/metadata/RuleProviderMetadata.java [hunk 3]
```diff
@@ -69,4 +69,22 @@
...
     public List<String> getExecuteBeforeIDs();
+
+
+    /**
+     * Whether Windup should stop execution if this provider's rule execution ends with an exception.
+     *
+     * By default, the exceptions are only logged and the failing rule appears in report.
+     * The rule itself is responsible for handling exceptions and storing them into the graph.
+     */
+    public void setHaltOnException(boolean haltOnException);
```
Comment (jsight) [inline config/api/src/main/java/org/jboss/windup/config/metadata/RuleProviderMetadata.java]: I don't think a setter belongs here.

File: config/api/src/main/java/org/jboss/windup/config/operation/Iteration.java [hunk 4]
```diff
@@ -83,7 +83,7 @@ private Iteration(FramesSelector selectionManager)
 
     /**
      * Begin an {@link Iteration} over the named selection of the given type. Also sets the name and type of the
-     * variable for this iteration's "current element". The type server for automatic type check.
+     * variable for this iteration's "current element". The type serves for automatic type check.
```
Comment (jsight) [inline config/api/src/main/java/org/jboss/windup/config/operation/Iteration.java]: Good.

File: config/api/src/main/java/org/jboss/windup/config/metadata/RuleProviderMetadata.java (hunk 3 above)
Comment (OndraZizka) [inline config/api/src/main/java/org/jboss/windup/config/metadata/RuleProviderMetadata.java]: Right. Will fix.

File: config/api/src/main/java/org/jboss/windup/config/metadata/RuleMetadataType.java (hunk 2 above)
Comment (OndraZizka) [inline config/api/src/main/java/org/jboss/windup/config/metadata/RuleMetadataType.java]: Right. Will fix.

File: config/api/src/main/java/org/jboss/windup/config/metadata/MetadataBuilder.java (hunk 1 above)
Comment (OndraZizka) [inline config/api/src/main/java/org/jboss/windup/config/metadata/MetadataBuilder.java]: Right. Will fix.

**Comment by windupgithubbot1** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://10.3.8.5//job/windup-pr-builder/611/
Test PASSed.

File: config/api/src/main/java/org/jboss/windup/config/metadata/RuleMetadataType.java (hunk 2 above)
Comment (OndraZizka) [inline config/api/src/main/java/org/jboss/windup/config/metadata/RuleMetadataType.java]: :sparkles:

File: config/api/src/main/java/org/jboss/windup/config/operation/Iteration.java (hunk 4 above)
Comment (OndraZizka) [inline config/api/src/main/java/org/jboss/windup/config/operation/Iteration.java]: :ok_hand:

**Comment by windupgithubbot1** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://10.3.8.5//job/windup-pr-builder/613/
Test PASSed.

**Comment by windupgithubbot1** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://10.3.8.5//job/windup-pr-builder/614/
Test PASSed.

**Pull request merged**